pkg/kube/container_oci.go

```diff
@@ -322,3 +321,50 @@ func (t *containerTranslator) configureAnnotations() {
 		t.g.AddAnnotation(k, v)
 	}
 }
+
+func (t *containerTranslator) configureUser() error {
+	security := t.cont.GetLinux().GetSecurityContext()
+	var userParts []string
+	if security.GetRunAsUsername() != "" {
+		userParts = append(userParts, security.GetRunAsUsername())
+	}
+	if security.GetRunAsUser() != nil {
+		userParts = append(userParts, fmt.Sprintf("%d", security.GetRunAsUser().GetValue()))
+	}
+	if security.GetRunAsGroup() != nil {
+		userParts = append(userParts, fmt.Sprintf("%d", security.GetRunAsGroup().GetValue()))
+	}
+
+	userSpec := strings.Join(userParts, ":")
+	containerUser, err := getContainerUser(t.cont.rootfsPath(), userSpec)
+	if err != nil {
+		return err
+	}
+
+	t.g.SetProcessUID(uint32(containerUser.Uid))
+	t.g.SetProcessGID(uint32(containerUser.Gid))
+	for _, gid := range containerUser.Sgids {
+		t.g.AddProcessAdditionalGid(uint32(gid))
+	}
+	for _, gid := range security.GetSupplementalGroups() {
```

Comment:
Additional groups are not checked ?

Reply:
No, they are simply added. Only primary group is validated (if present)